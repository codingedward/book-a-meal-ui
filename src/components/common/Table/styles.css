.table-holder {
  width: 100%;
  overflow-x: scroll;
  padding: 0;
  margin: 0; }
