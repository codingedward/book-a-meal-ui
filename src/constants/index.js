export const BASE_URL = 'http://localhost:8000/api/v1';
export const IMAGES_UPLOAD_URL = 'https://api.cloudinary.com/v1_1/codingedward/image/upload';

export const Status = {
    DEFAULT: -1,
    STARTED: 0,
    SUCCESS: 1,
    FAIL: 2,

}

export const Role = {
    SUPER_ADMIN: 0,
    ADMIN: 1,
    USER: 2,
}

export  const EntryType = {
    TEXT: 0,
    NUMBER: 1,
    IMAGE: 2,
    DATE: 3,
}

