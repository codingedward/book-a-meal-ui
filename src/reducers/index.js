import { combineReducers } from 'redux';
import auth from './AuthReducer';
import meals from './MealsReducer';

const reducer = combineReducers({
    auth,
    meals
})

export default reducer;
