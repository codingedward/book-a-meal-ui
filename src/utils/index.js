import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';

export { singleError, paginationInfo, authenticated }

