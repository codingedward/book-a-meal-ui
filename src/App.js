import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';

import Landing from './components/views/Landing';
import Login from 'src/components/views/auth/Login';
import Register from 'src/components/views/auth/Register';
import Admin from 'src/components/views/admin';
import './styles.css';

class App extends React.Component {
    render() {
        return (
            <Router>
                <Switch>
                    <Route exact path="/" component={Landing} />
                    <Route path="/admin" component={Admin} />
                    <Route path="/login" component={Login} />
                    <Route path="/sign-up" component={Register} />
                </Switch>
            </Router>
        );
    }
}

export default App;
