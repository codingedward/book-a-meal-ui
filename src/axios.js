import axios from 'axios';
import { BASE_URL } from './constants';

axios.defaults.baseURL = BASE_URL;
axios.defaults.headers.common['Authorization'] = localStorage.getItem('token');

axios.auth = () => {
    axios.defaults.headers.common['Authorization'] = localStorage.getItem('token');
}


export default axios;

