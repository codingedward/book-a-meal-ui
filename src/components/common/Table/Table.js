import React from 'react';
import { EntryType } from '../../../constants';

const Table = ({ data, onDelete, onEdit }) => {

    const renderEntry = (value, type) => {
        switch (type) {
            case EntryType.IMAGE:
                return <img src={value} alt="Entry" />;
            case EntryType.DATE:
                return new Date(value).toDateString()
            case EntryType.TEXT:
            case EntryType.NUMBER:
            default:
                return value;
        }
    }

    return (
        <div className="table-holder">
            <table className="table">
                <thead>
                    <tr>
                        {data.columns.map(column => (
                            <th key={column.title}> {column.title} </th>
                        ))}

                        { /* editable? */ }
                        {onEdit && <th> Edit </th> }

                        { /* deletable? */ }
                        {onDelete && <th> Delete </th> }
                    </tr>
                </thead>
                <tbody>
                    {data.rows.map((row, rIndex) => (
                        <tr key={rIndex}>
                            {data.columns.map((column, cIndex) => (
                                <td key={`${rIndex}${cIndex}`}>
                                    { renderEntry(row[column.key], column.type) }
                                </td>
                            ))}

                            { /* editable? */ }
                            {onEdit &&
                                <td>
                                    <button onClick={() => onEdit()} className="edit-act">
                                        Edit 
                                    </button>
                                </td>
                            }

                            { /* deletable? */ }
                            {onDelete &&
                                <td>
                                    <button onClick={() => onDelete()} className="delete-act">
                                        Delete
                                    </button>
                                </td>
                            }
                        </tr> 
                    ))}
                </tbody>
            </table>
        </div>
    );
}

export default Table;
