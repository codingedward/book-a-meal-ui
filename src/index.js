import React from 'react';
import ReactDOM from 'react-dom';

import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';

import axios from 'axios';
import axiosMiddleware from 'redux-axios-middleware';

import { createBrowserHistory } from 'history';
import { 
    ConnectedRouter,
    connectRouter, 
    routerMiddleware 
} from 'connected-react-router';

import App from './App';
import appReducer from './reducers';
import registerServiceWorker from './registerServiceWorker';

const history = createBrowserHistory();

const client = axios.create({
    baseURL: 'http://localhost:5000/api/v1',
    responseType: 'json',
});

const store = createStore(
    connectRouter(history)(appReducer),
    applyMiddleware(axiosMiddleware(client), routerMiddleware(history))
);

ReactDOM.render(
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <App/>
        </ConnectedRouter>
    </Provider>,
    document.getElementById('root')
);
registerServiceWorker();
