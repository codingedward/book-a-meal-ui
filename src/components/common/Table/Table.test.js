import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; // order matters!
import Table from './Table';
import Adapter from 'enzyme-adapter-react-16';
import { EntryType } from 'src/constants';

const data = {
    columns: [
        {
            type: EntryType.TEXT,
            key: 'name',
            title: 'Name',
        }
    ],
    rows: [ { name: 'John Doe' } ]
}

describe('<Table />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Table data={data} />);
        expect(wrapper).toMatchSnapshot();
    });
});

