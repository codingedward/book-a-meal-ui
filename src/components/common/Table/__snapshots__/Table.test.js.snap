// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    data={
        Object {
            "columns": Array [
              Object {
                "key": "name",
                "title": "Name",
                "type": 0,
              },
            ],
            "rows": Array [
              Object {
                "name": "John Doe",
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <table
          className="table"
>
          <thead>
                    <tr>
                              <th>
                                         
                                        Name
                                         
                              </th>
                    </tr>
          </thead>
          <tbody>
                    <tr>
                              <td>
                                        John Doe
                              </td>
                    </tr>
          </tbody>
</table>,
        false,
      ],
      "className": "table-holder",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                            <th>
                                           
                                          Name
                                           
                            </th>
              </tr>
</thead>,
            <tbody>
              <tr>
                            <td>
                                          John Doe
                            </td>
              </tr>
</tbody>,
          ],
          "className": "table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                                 
                                Name
                                 
                </th>
</tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    <th>
                       
                      Name
                       
</th>,
                  ],
                  undefined,
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Name",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Name",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Name",
                    " ",
                  ],
                  "type": "th",
                },
                undefined,
                undefined,
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                                    John Doe
                  </td>
</tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    Array [
                      <td>
                        John Doe
</td>,
                    ],
                    undefined,
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "00",
                    "nodeType": "host",
                    "props": Object {
                      "children": "John Doe",
                    },
                    "ref": null,
                    "rendered": "John Doe",
                    "type": "td",
                  },
                  undefined,
                  undefined,
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <table
            className="table"
>
            <thead>
                        <tr>
                                    <th>
                                                 
                                                Name
                                                 
                                    </th>
                        </tr>
            </thead>
            <tbody>
                        <tr>
                                    <td>
                                                John Doe
                                    </td>
                        </tr>
            </tbody>
</table>,
          false,
        ],
        "className": "table-holder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                                <th>
                                                 
                                                Name
                                                 
                                </th>
                </tr>
</thead>,
              <tbody>
                <tr>
                                <td>
                                                John Doe
                                </td>
                </tr>
</tbody>,
            ],
            "className": "table",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                                     
                                    Name
                                     
                  </th>
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    Array [
                      <th>
                         
                        Name
                         
</th>,
                    ],
                    undefined,
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "Name",
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Name",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Name",
                      " ",
                    ],
                    "type": "th",
                  },
                  undefined,
                  undefined,
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr>
                    <td>
                                        John Doe
                    </td>
</tr>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      Array [
                        <td>
                          John Doe
</td>,
                      ],
                      undefined,
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "00",
                      "nodeType": "host",
                      "props": Object {
                        "children": "John Doe",
                      },
                      "ref": null,
                      "rendered": "John Doe",
                      "type": "td",
                    },
                    undefined,
                    undefined,
                  ],
                  "type": "tr",
                },
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
